app.go

```diff
@@ -5,41 +5,70 @@ import (
 	"net/http"
 
 	"github.com/gorilla/mux"
+	"github.com/jmoiron/sqlx"
 	"github.com/justinas/alice"
+	_ "github.com/lib/pq"
 
 	"github.com/makeict/MESSforMakers/controllers"
+	"github.com/makeict/MESSforMakers/models"
 	"github.com/makeict/MESSforMakers/session"
 	"github.com/makeict/MESSforMakers/util"
 )
 
 // database connection, cookie store, etc..
 type application struct {
-	cookieStore    *session.CookieStore
-	commonHandlers alice.Chain
-	logger         *util.Logger
+	cookieStore *session.CookieStore
+	logger      *util.Logger
+	DB          *sqlx.DB
+	Router      http.Handler
 }
 
 func newApplication() *application {
+
+	//Set up a logger middleware
 	logger, err := util.NewLogger()
 	if err != nil {
 		fmt.Printf("Error creating logger :: %v", err)
 		panic(1)
 	}
-	loggingMiddleware := loggingMiddleware{true, logger}
+	loggingMiddleware := loggingMiddleware{true, logger} // currently defaults dump_request to true, needs to be a config option
+
+	//middlewware that should be called on every request get added to the chain here
+	commonHandlers := alice.New(loggingMiddleware.loggingHandler)
+
+	//set up the database
+	db, err := models.InitDB("postgres://vztngihfamevpx:2555f8b0e9f19cef182deb741958401065d75e1fe6aa91839a383908e9717682@ec2-23-21-220-32.compute-1.amazonaws.com:5432/dc9hak1f3kqc35")
```

Comment:
Code should read Postgres username and password from environment variable or config. Password and username should not be committed to git.

Reply:
Yeah, I had that as a generic "usernamehere:passwordhere" but then went back and tweaked something and forgot to switch it back. Thanks for catching.  That's also why I was starting to work on the config.json